src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -830,6 +830,19 @@ private void ParseProject(string firstLine)
                         line = ReadLine();
                     }
                 }
+                else if (line.StartsWith("Project(", StringComparison.Ordinal))
+                {
+                    // Another Project spotted instead of EndProject for the current one - solution file is malformed
+                    string warning = ResourceUtilities.FormatResourceStringStripCodeAndKeyword(out _, out _, "Shared.InvalidProjectFile",
+                        _solutionFile, proj.ProjectName);
+                    SolutionParserWarnings.Add(warning);
```

Comment:
Not sure if you intended this as a warning that would interfere with the build or not. It doesn't, which is good.

Reply:
It was intended to be only the warning - we want to keep the build running fine (just like it is when building this case in Visual Studio) but we still want to keep the developer/user notified that the solution file is malformed.